hydra-node/test/Hydra/Port/ServerSpec.hs

```diff
@@ -0,0 +1,339 @@
+{-# LANGUAGE TypeApplications #-}
+
+module Hydra.Port.ServerSpec (spec) where
+
+import Hydra.Prelude hiding (decodeUtf8, seq)
+import Test.Hydra.Prelude
+
+import Cardano.Binary (serialize')
+import Control.Exception (IOException)
+import Control.Lens ((^?))
+import Control.Monad.Class.MonadSTM (
+  check,
+  modifyTVar',
+  newTQueue,
+  newTVarIO,
+  readTQueue,
+  readTVarIO,
+  tryReadTQueue,
+  writeTQueue,
+ )
+import qualified Data.Aeson as Aeson
+import qualified Data.Aeson.KeyMap as KeyMap
+import Data.Aeson.Lens (key, nonNull)
+import qualified Data.ByteString.Base16 as Base16
+import qualified Data.Text as T
+import Data.Text.Encoding (decodeUtf8)
+import Hydra.API.Server (Server (Server, sendOutput), withAPIServer)
+import Hydra.API.ServerOutput (ServerOutput (..), TimedServerOutput (..), input)
+import Hydra.Chain (HeadId (HeadId), PostChainTx (CloseTx), PostTxError (NoSeedInput), confirmedSnapshot)
+import Hydra.Ledger.Simple (SimpleTx)
+import Hydra.Logging (nullTracer, showLogsOnFailure)
+import Hydra.Network (PortNumber)
+import Hydra.Persistence (PersistenceIncremental (..), createPersistenceIncremental)
+import Hydra.Snapshot (ConfirmedSnapshot (..), Snapshot, confirmed)
+import Network.WebSockets (Connection, receiveData, runClient, sendBinaryData)
+import System.Timeout (timeout)
+import Test.Hydra.Fixture (alice)
+import Test.Network.Ports (withFreePort)
+import Test.QuickCheck (checkCoverage, cover, generate)
+import Test.QuickCheck.Monadic (monadicIO, monitor, pick, run)
+
+spec :: SpecWith (TVar IO [Int])
+spec =
+  describe "ServerSpec" $
+    parallel $ do
+      it "greets" $ \tvar ->
```

Comment:
Must: not litter the code with anonymous variables

The TVar littering here and above in the signature is to be avoided at all costs. It makes the code harder to read and impossible to know (without context) what this `tvar` is for? Even the type is very anonymous.